ts/main.ts

```diff
@@ -183,13 +194,15 @@ async function fetchExerciseSearchData(term: string): Promise<void> {
     }
     if (exerciseObjArr.length > 0) {
       exerciseObjArr.forEach((element) => {
-        $cardList.appendChild(renderExercises(element));
+        $exercisesCardList.appendChild(renderExercises(element));
       });
       $noResults?.classList.add('hidden');
     } else {
       $noResults?.classList.remove('hidden');
     }
-    $exercisesNodeList = document.querySelectorAll('.card');
+    const $exercisesNodeList = document.querySelectorAll(
+      '#exercises-card-list > .card',
+    );
     if (!$exercisesNodeList) throw new Error('no exercise nodelist found');
```

Comment:
Do you need this here? This doesn't look like it's doing anything. You added a query for it in your `handleFavoriteClick` function where it's used, but you query here, throw an error if it returns null but why?